web_src/js/modules/worker.js

```diff
@@ -0,0 +1,8 @@
+const {appSubUrl} = window.config;
+
+export function logoutFromWorker() {
+  // wait for a while because other requests (eg: logout) may be in the flight
+  setTimeout(() => {
+    window.location.href = `${appSubUrl}/`;
+  }, 5000);
```

Comment:
That looks really hacky: Instead of logging out immediately, we always wait for five seconds instead?
That doesn't sound like a good idea.
Instead, what about keeping track of which `fetch`es are still active and shutting down once the last one completed?

Reply:
Although it looks hacky, it is quite reliable and I believe it is good enough. This "five seconds" usually does nothing, because the current page should already been redirected by the "logout" action. It only take affect if there is a background tab, and the user signed out from another tab. Then the 5 seconds is also good for a background tab.

`keeping track of which fetch` will be much more hacky, and introduce unnecessary changes and untestable code, it is not suitable to backport.

Reply:
You could take a try, it always "logs out immediately", no "waits for 5 seconds".